qa/tasks/vstart_runner.py

```diff
@@ -394,41 +394,42 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
-    # XXX: omit_sudo is re-set to False even in cases of commands like passwd
-    # and chown.
-    # XXX: "adjust-ulimits", "ceph-coverage" and "sudo" in command arguments
-    # are overridden. Former two usually have no applicability for test runs
-    # on developer's machines and see note point on "omit_sudo" to know more
-    # about overriding of "sudo".
-    # XXX: the presence of binary file named after the first argument is
-    # checked in build/bin/, if present the first argument is replaced with
-    # the path to binary file.
-    def _perform_checks_and_adjustments(self, args, omit_sudo, shell):
-        if isinstance(args, list):
-            args = quote(args)
-
-        assert isinstance(args, str)
-
-        first_arg = args[ : args.find(' ')]
-        if '/' not in first_arg:
-            local_bin = os.path.join(BIN_PREFIX, first_arg)
-            if os.path.exists(local_bin):
-                args = args.replace(first_arg, local_bin, 1)
+    def _omit_cmd_args(self, args, omit_sudo):
+        """
+        Helper tools are omitted since those are not meant for tests executed
+        using vstart_runner.py. And sudo's omission depends on the value of
+        the variable omit_sudo.
+        """
+        helper_tools = ('adjust-ulimits', 'ceph-coverage',
+                        'None/archive/coverage')
+        for i in helper_tools:
+            if i in args:
+                helper_tools_found = True
+                break
+        else:
+            helper_tools_found = False
 
-        # we're intentionally logging the command before overriding to avoid
-        # this clutter in the log.
-        log.debug('> ' + args)
+        if not helper_tools_found and not 'sudo' in args:
```

Comment:
do you mean `if not helper_tools_found and 'sudo' not in args`?

Reply:
Rest LGTM

Reply:
Yep, fixed.